src/azure-cli/azure/cli/command_modules/aro/_params.py

```diff
@@ -0,0 +1,91 @@
+
+from azure.cli.command_modules.aro._validators import validate_cidr
+from azure.cli.command_modules.aro._validators import validate_client_id
+from azure.cli.command_modules.aro._validators import validate_client_secret
+from azure.cli.command_modules.aro._validators import validate_cluster_resource_group
+from azure.cli.command_modules.aro._validators import validate_domain
+from azure.cli.command_modules.aro._validators import validate_pull_secret
+from azure.cli.command_modules.aro._validators import validate_subnet
+from azure.cli.command_modules.aro._validators import validate_visibility
+from azure.cli.command_modules.aro._validators import validate_vnet
+from azure.cli.command_modules.aro._validators import validate_vnet_resource_group_name
+from azure.cli.command_modules.aro._validators import validate_worker_count
+from azure.cli.command_modules.aro._validators import validate_worker_vm_disk_size_gb
+from azure.cli.core.commands.parameters import name_type
+from azure.cli.core.commands.parameters import resource_group_name_type
+from azure.cli.core.commands.parameters import tags_type
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+
+
+def load_arguments(self, _):
+    with self.argument_context('aro') as c:
+        c.argument('location',
+                   validator=get_default_location_from_resource_group)
+        c.argument('resource_name',
+                   name_type,
+                   help='Name of cluster.')
+        c.argument('tags',
+                   tags_type)
+
+        c.argument('pull_secret',
+                   help='Pull secret of cluster.',
+                   validator=validate_pull_secret)
+        c.argument('domain',
+                   help='Domain of cluster.',
+                   validator=validate_domain)
+        c.argument('cluster_resource_group',
+                   help='Resource group of cluster.',
+                   validator=validate_cluster_resource_group)
+
+        c.argument('client_id',
+                   help='Client ID of cluster service principal.',
+                   validator=validate_client_id)
+        c.argument('client_secret',
+                   help='Client secret of cluster service principal.',
+                   validator=validate_client_secret)
+
+        c.argument('pod_cidr',
+                   help='CIDR of pod network.',
+                   validator=validate_cidr('pod_cidr'))
+        c.argument('service_cidr',
+                   help='CIDR of service network.',
+                   validator=validate_cidr('service_cidr'))
+
+        c.argument('master_vm_size',
+                   help='Size of master VMs.')
+
+        c.argument('worker_vm_size',
+                   help='Size of worker VMs.')
+        c.argument('worker_vm_disk_size_gb',
+                   help='Disk size in GB of worker VMs.',
+                   validator=validate_worker_vm_disk_size_gb)
```

Comment:
You can add `type=int` to require the parameter value is an integer to save the work of validate function `_validate_int`

Reply:
Will add this as follow-up issue[1] to our codebase, as all this code is a copy of our extension we use for PrivatePreview and development (https://github.com/Azure/ARO-RP/tree/master/python/az/aro). If this is ok, will fix in follow-up PR.

1. https://github.com/Azure/ARO-RP/issues/368